hardware_interface/include/hardware_interface/handle.hpp

```diff
@@ -139,6 +328,118 @@ class CommandInterface : public ReadWriteHandle
   using ReadWriteHandle::ReadWriteHandle;
 };
 
+class DistributedReadWriteHandle : public ReadWriteHandle
+{
+public:
+  DistributedReadWriteHandle(
+    const distributed_control::PublisherDescription & description, const std::string & ns = "/")
+  : ReadWriteHandle(description.prefix_name(), description.interface_name(), &value_),
+    get_value_topic_name_(description.topic_name()),
+    namespace_(ns),
+    interface_namespace_(description.get_namespace()),
+    forward_command_topic_name_(get_underscore_separated_name() + "_command_forwarding")
+  {
+    // create node for subscribing to StatePublisher described in StatePublisherDescription
+    rclcpp::NodeOptions node_options;
+    node_ = std::make_shared<rclcpp_lifecycle::LifecycleNode>(
+      get_underscore_separated_name() + "_distributed_command_interface", namespace_, node_options,
+      false);
+
+    // subscribe to topic provided by StatePublisher
+    command_value_sub_ = node_->create_subscription<std_msgs::msg::Float64>(
+      get_value_topic_name_, 10,
+      std::bind(&DistributedReadWriteHandle::get_value_cb, this, std::placeholders::_1));
+
+    // create publisher so that we can forward the commands
+    command_value_pub_ =
+      node_->create_publisher<std_msgs::msg::Float64>(forward_command_topic_name_, 10);
+  }
+
+  explicit DistributedReadWriteHandle(const std::string & interface_name)
+  : ReadWriteHandle(interface_name)
+  {
+  }
+
+  explicit DistributedReadWriteHandle(const char * interface_name) : ReadWriteHandle(interface_name)
+  {
+  }
+
+  DistributedReadWriteHandle(const DistributedReadWriteHandle & other) = default;
+
+  DistributedReadWriteHandle(DistributedReadWriteHandle && other) = default;
+
+  DistributedReadWriteHandle & operator=(const DistributedReadWriteHandle & other) = default;
+
+  DistributedReadWriteHandle & operator=(DistributedReadWriteHandle && other) = default;
+
+  virtual ~DistributedReadWriteHandle() = default;
+
+  virtual std::string get_name() const override
+  {
+    // concatenate: interface_namespace/prefix_name/interface_name to obtain
+    // a unique name.
+    return append_char(interface_namespace_, '/') + append_char(get_prefix_name(), '/') +
+           get_interface_name();
+  }
+
+  virtual std::string get_underscore_separated_name() const override
+  {
+    // remove first "/" from namespace and replace all follow occurrences with "_"
```

Comment:
why do we need this?

Reply:
Probably needed for node names. Why not use one node name per CM/RM and namespaces, e.g. /<controller_manager>/<hardware>/<joint>/<interface>